Kernel/SpinLock.h

```diff
@@ -126,22 +126,23 @@ class RecursiveSpinLock {
 };
 
 template<typename LockType>
-class ScopedSpinLock {
+class [[nodiscard]] ScopedSpinLock
+{
     AK_MAKE_NONCOPYABLE(ScopedSpinLock);
 
 public:
     ScopedSpinLock() = delete;
     ScopedSpinLock& operator=(ScopedSpinLock&&) = delete;
 
-    ScopedSpinLock(LockType& lock)
+    ScopedSpinLock(LockType & lock)
```

Comment:
:thinking: that be weird.
I do remember clang-format having issues when classes have [[attributes]], maybe `// clang-format off` the class decl?

Reply:
Sure, I can try... auto formatters are great until they go haywire 😁🔫

Reply:
I just added a new NO_DISCARD macro to be able to  hide this behind, clang-format off didn't work, and still mangled the rest of the file.